instrumentation-api-semconv/src/testStableHttpSemconv/java/io/opentelemetry/instrumentation/api/instrumenter/net/NetClientAttributesExtractorStableSemconvTest.java

```diff
@@ -0,0 +1,138 @@
+        Map<String, String> request, @Nullable Map<String, String> response) {
+      return request.get("protocolVersion");
+    }
+
+    @Override
+    public String getPeerName(Map<String, String> request) {
+      return request.get("peerName");
+    }
+
+    @Override
+    public Integer getPeerPort(Map<String, String> request) {
+      String peerPort = request.get("peerPort");
+      return peerPort == null ? null : Integer.valueOf(peerPort);
+    }
+
+    @Override
+    public String getSockFamily(Map<String, String> request, Map<String, String> response) {
+      return response.get("sockFamily");
+    }
+
+    @Override
+    public String getSockPeerAddr(Map<String, String> request, Map<String, String> response) {
+      return response.get("sockPeerAddr");
+    }
+
+    @Override
+    public String getSockPeerName(Map<String, String> request, Map<String, String> response) {
+      return response.get("sockPeerName");
+    }
+
+    @Override
+    public Integer getSockPeerPort(Map<String, String> request, Map<String, String> response) {
+      String sockPeerPort = response.get("sockPeerPort");
+      return sockPeerPort == null ? null : Integer.valueOf(sockPeerPort);
+    }
+  }
+
+  private final AttributesExtractor<Map<String, String>, Map<String, String>> extractor =
+      NetClientAttributesExtractor.create(new TestNetClientAttributesGetter());
+
+  @Test
+  void normal() {
+    // given
+    Map<String, String> map = new HashMap<>();
+    map.put("netTransport", IP_TCP);
+    map.put("transport", "tcp");
+    map.put("type", "ipv6");
+    map.put("protocolName", "http");
+    map.put("protocolVersion", "1.1");
+    map.put("peerName", "opentelemetry.io");
+    map.put("peerPort", "42");
+    map.put("sockFamily", "inet6");
+    map.put("sockPeerAddr", "1:2:3:4::");
+    map.put("sockPeerName", "proxy.opentelemetry.io");
+    map.put("sockPeerPort", "123");
+
+    Context context = Context.root();
+
+    // when
+    AttributesBuilder startAttributes = Attributes.builder();
+    extractor.onStart(startAttributes, context, map);
+
+    AttributesBuilder endAttributes = Attributes.builder();
+    extractor.onEnd(endAttributes, context, map, map, null);
+
+    // then
+    assertThat(startAttributes.build())
+        .containsOnly(
+            entry(SemanticAttributes.NET_PEER_NAME, "opentelemetry.io"),
+            entry(SemanticAttributes.NET_PEER_PORT, 42L));
+
+    assertThat(endAttributes.build())
+        .containsOnly(
+            entry(NetworkAttributes.NETWORK_TRANSPORT, "tcp"),
+            entry(NetworkAttributes.NETWORK_TYPE, "ipv6"),
+            entry(NetworkAttributes.NETWORK_PROTOCOL_NAME, "http"),
+            entry(NetworkAttributes.NETWORK_PROTOCOL_VERSION, "1.1"),
+            entry(SemanticAttributes.NET_SOCK_PEER_ADDR, "1:2:3:4::"),
+            entry(SemanticAttributes.NET_SOCK_PEER_NAME, "proxy.opentelemetry.io"),
+            entry(SemanticAttributes.NET_SOCK_PEER_PORT, 123L));
```

Comment:
ah, I see, these will be part of next PR?

I'm guessing that explains my questions about `InternalNetServerAttributesExtractor` and `InternalNetClientAttributesExtractor`?

Reply:
Yup, exactly that!